addons/website/models/website_visitor.py

```diff
@@ -28,14 +29,30 @@ class WebsiteTrack(models.Model):
 class WebsiteVisitor(models.Model):
     _name = 'website.visitor'
     _description = 'Website Visitor'
-    _order = 'last_connection_datetime DESC'
+    _order = 'id DESC'
 
-    name = fields.Char('Name')
-    access_token = fields.Char(required=True, default=lambda x: uuid.uuid4().hex, copy=False, groups='website.group_website_publisher')
-    active = fields.Boolean('Active', default=True)
+    def _get_access_token(self):
+        """ Either the user's partner.id or a hash. """
+        assert request, _("Visitors can only be created through the frontend.")
+        if not self.env.user._is_public():
+            return self.env.user.partner_id.id
+        # TODO: Useless to use secret?
+        secret = self.env["ir.config_parameter"].sudo().get_param("database.secret")
+        msg = repr((
+            # For testing purpose, uncomment this line if you want different
+            # URL (domain/ip) to generate different visitor.
+            # request.httprequest.host,
+            request.httprequest.remote_addr,
+            # See commit message. Once PR is validated, check how to handle request when no user_agent
+            request.httprequest.environ['HTTP_USER_AGENT'],
+            fields.Date.today()
+        )).encode('utf-8')
+        return 'tok_' + hmac.new(secret.encode('utf-8'), msg, hashlib.sha1).hexdigest()
```

Comment:
What about removing "tok_"? Just to save time and disk space. (website.visitor can be a very large table; no good reason to add 4 characters per row: 54M records for odoo.com, that means 216Mb saved). You might use len() instead of is_digit.